compiler/parser/ztests/template-literals.yaml

```diff
@@ -0,0 +1,26 @@
+zed: |
+  over this
+  | yield
+    "hello ${this}",
+    'hello ${to_hex(this)}',
+    "hello \${this}",
+    'hello \${to_hex(this)}',
+    'yo ${"dawg ${this}"}',
+    "==="
+
+input: |
+  "world" 127.0.0.1
+
+output: |
+   "hello world"
+   "hello 776f726c64"
+   "hello ${this}"
+   "hello ${to_hex(this)}"
+   "yo dawg world"
+   "==="
+   "incompatible types"(error)
+   "hello 7f000001"
+   "hello ${this}"
+   "hello ${to_hex(this)}"
+   "yo incompatible types"(bstring)
```

Comment:
This feels like a bug, but not from this pr.

Reply:
> This feels like a bug, but not from this pr.

This should be fixed in #3396